plugins/remote-config/frontend/public/javascripts/countly.views.js

```diff
@@ -13,51 +18,247 @@
+                    }
+                    if (typeof (default_value) === 'object') {
+                        doc.default_value = JSON.stringify(doc.default_value);
+                    }
                     this.title = "Update parameter";
                     this.saveButtonLabel = "Save";
+                    if (doc.description) {
+                        this.showDescription = true;
+                    }
+                    if (doc.valuesList) {
+                        this.valuesList = doc.valuesList;
+                    }
+                    if (doc.expiry_dttm) {
+                        this.showExpirationDate = true;
+                    }
                 }
                 else {
                     this.title = "Add parameter";
                     this.saveButtonLabel = "Save";
+                    this.showDescription = false;
+                    this.valuesList = [];
+                    this.showExpirationDate = false;
                 }
+            },
+            openJsonEditor: function() {
+                this.$store.dispatch("countlyRemoteConfig/parameters/showJsonEditor", true);
             }
         }
     });
 
+    var ColorTag = countlyVue.views.BaseView.extend({
+        data: function() {
+            return {
+                selectedColorKey: this.defaultColorKey
+            };
+        },
+        methods: {
+            click: function(event) {
+                this.selectedColorKey = event.target.getAttribute('data-color');
+                this.$emit("input", Number(this.selectedColorKey));
+            },
+        },
+
+        props: {
+            value: Number,
+            tags: {
+                type: Array,
+                default: []
+            },
+            defaultColorKey: {
+                type: Number,
+                default: 1
+            }
+        },
+        template: '<div class="bu-is-flex bu-is-flex-wrap-wrap bu-is-align-items-center">\
+                                <div class="cly-vue-remote-config-conditions-drawer__color-tag-wrapper"  v-for="(tag,idx) in tags" :data-color="tag.key">{{ tag.label }}\
+                                <div v-if="tag.key == selectedColorKey" @click="click" class="cly-vue-remote-config-conditions-drawer__color-tag cly-vue-remote-config-conditions-drawer__color-tag__selected bu-is-flex bu-is-align-items-center bu-is-justify-content-center" :style="{backgroundColor: tag.colorCode}" :data-color="tag.key">\
+                                    <i class="ion-checkmark cly-vue-remote-config-conditions-drawer__checkmark"></i>\
+                                </div>\
+                                <div v-else @click="click" class="cly-vue-remote-config-conditions-drawer__color-tag bu-is-flex bu-is-align-items-center bu-is-justify-content-center" :style="{backgroundColor: tag.colorCode}" :data-color="tag.key"></div>\
+                                </div>\
+                    </div>'
+    });
+    var JsonEditor = countlyVue.views.create({
```

Comment:
This should be made a global component and kept in .../vue/components/helpers.js
Please move its template to core as well in .../vue/components/templates

Reply:
Also please modify this component a bit.
json editor should be an independent component that can be used within cly-dialog or any other component.

Currently, json editor is a part of cly-dialog, so if someone wants to use json editor without the cly-dialog, they cannot do so.